AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/RemoteSyncEngine+State.swift

```diff
@@ -0,0 +1,53 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import Combine
+@available(iOS 13.0, *)
+extension RemoteSyncEngine {
+
+    /// States are descriptive, they say what is happening in the system right now
+    enum State {
+        case notStarted
+
+        case pauseSubscriptions
+        case pauseMutationQueue
+        case initializeSubscriptions(APICategoryGraphQLBehavior, StorageEngineAdapter)
+        case performInitialSync
+        case activateCloudSubscriptions
+        case activateMutationQueue(APICategoryGraphQLBehavior, MutationEventPublisher)
+        case notifySyncStarted
```

Comment:
nit: states should describe what is happening. Consider changing to gerund phrases to indicate ongoing activity (e.g., `pausingSubscriptions`) or declarative phrases to indicate something more long-lived (like you already have with, `syncEngineActive`). Disclaimer: I am not a grammarist, so I may be using the wrong names here. :)

Reply:
Updated: https://github.com/aws-amplify/amplify-ios/pull/329